robottelo/api/utils.py

```diff
@@ -940,42 +940,42 @@ def create_org_admin_user(orgs, locs):
     return user
 
 
-def skip_yum_update_during_provisioning(template=None, reverse=False):
+@contextmanager
+def skip_yum_update_during_provisioning(template=None):
     """Hides the yum update command with echo text
 
     :param str template: The template name where the yum update will be hidden
-    :param bool reverse: Reverses the echo text to yum update
-    :return: Boolean True on success else exception
     """
     old = 'yum -t -y update'
     new = 'echo "Yum update skipped for faster automation testing"'
-    if not reverse:
-        return update_provisioning_template(name=template, old=old, new=new)
-    else:
-        return update_provisioning_template(name=template, old=new, new=old)
+    update_provisioning_template(name=template, old=old, new=new)
+    yield
+    update_provisioning_template(name=template, old=new, new=old)
 
 
-def set_hammer_api_timeout(timeout=-1, reverse=False):
+@contextmanager
+def hammer_api_timeout(timeout=-1):
     """Set hammer API request timeout on Satellite
 
     :param int timeout: request timeout in seconds
-    :param bool reverse: Reverses the request timeout
-    :return: ssh.command
     """
-    default_timeout = f':request_timeout: {120}'
     new_timeout = f':request_timeout: {timeout}'
-    if not reverse:
-        return ssh.command(
-            "sed -ie 's/{}/{}/' ~/.hammer/cli.modules.d/foreman.yml".format(
-                default_timeout, new_timeout
-            )
-        )
+    cli_conf = "~/.hammer/cli.modules.d/foreman.yml"
+
+    if ssh.command(f"grep -i 'request_timeout' {cli_conf}").return_code != 0:
+        ssh.command(f"echo '  {new_timeout}' >> {cli_conf}")
```

Comment:
I think we can use `default_sat.execute` instead of `ssh.command` to run the command on satellite.

Reply:
Agreed, and this was discussed above. This isn't in a pytest fixture or test case, so the satellite instance will have to be passed down.  Its not a strict requirement to the PR, just makes for additional instances that @JacobCallahan or I have to replace in the unstable branch before it can be merged to master.

Reply:
In fact, this whole method should get moved to the `Capsule` class. Then its references to `self.execute`

Reply:
Either that or converted to a fixture. I think either of those options are better than just leaving it as an `api.utils` function that needs to be imported. The context manager pattern is nice to have, and I think indicates that this should be a fixture. Its only uses are in the azure test module.

Reply:
@mshriver Its not really Azure specific, this is a temporary update to satellite Hammer config that changes its timeout so that satellite hammer command wont fail with `TimeoutError`. This context manager can be used everywhere in a satellite like `Content View publish`, `Repo Sync` and `Provisioing(Azure, EC2, VMWare etc)`. The customers also been asked to adjust this timeout as per their environment.

And hence I am moving this under `Satellite` class as a context manager. Along with this `skip_yum_update_during_provisioning` is also going into satellite class.